dev/com.ibm.ws.security.wim.adapter.ldap_fat.krb5/fat/src/com/ibm/ws/security/wim/adapter/ldap/fat/krb5/ApacheDSandKDC.java

```diff
@@ -653,17 +653,44 @@ public static void createKeyTabFile() throws Exception {
 
         Log.info(c, "createKeyTabFile", "Created keytab: " + keytabFile);
         Log.info(c, "createKeyTabFile", "Keytab actual contents: " + FileUtils.readFile(keytabFile));
+    }
+
+    /**
+     * @param entries
+     */
+    protected static List<KeytabEntry> addKerberosKeysToKeytab(String principalName, String principalPwd) {
+        List<KeytabEntry> entries = new ArrayList<KeytabEntry>();
+        KerberosTime timeStamp = new KerberosTime();
+
+        addKerberosKeys(principalName, principalPwd, entries, timeStamp);
+
         // KeytabEntry doesn't have a nice toString
         for (KeytabEntry key : entries) {
             StringBuffer strBuf = new StringBuffer();
             strBuf.append(" PrincipalName: " + key.getPrincipalName());
-            strBuf.append(" PrincipalType: " + key.getPrincipalType());
+            //strBuf.append(" PrincipalType: " + key.getPrincipalType());
             strBuf.append(" Key: " + key.getKey());
             strBuf.append(" KeyVersion: " + key.getKeyVersion());
-            strBuf.append(" TimeStamp: " + key.getTimeStamp());
+            //strBuf.append(" TimeStamp: " + key.getTimeStamp());
 
-            Log.info(c, "createKeyTabFile", "Keytab entry: " + strBuf.toString());
+            Log.info(c, "addKerberosKeysToKeytab", "Keytab entry: " + strBuf.toString());
         }
 
+        return entries;
+    }
+
+    /**
+     * @param principalName
+     * @param principalPwd
+     * @param entries
+     */
+    private static void addKerberosKeys(String principalName, String principalPwd, List<KeytabEntry> entries, KerberosTime timeStamp) {
```

Comment:
Again, seems misnamed. This adds keytab entries to the existing list of keytab entries.